pkg/beacon/client/client.go

```diff
@@ -137,6 +137,18 @@ func (b *beaconClient) Genesis() (structs.GenesisInfo, error) {
 	return resp.Data, err
 }
 
+func (b *beaconClient) Randao(slot structs.Slot) (string, error) {
+	resp := new(GetRandaoResponse)
+	u := *b.beaconEndpoint
```

Comment:
wouldn't it be easier to store url.URL instead of pointer, than dereferencing this in every function? Asking for a friend....